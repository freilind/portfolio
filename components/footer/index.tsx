import React, { FC } from 'react';

const Footer: FC = () => {
    return (
        <footer>
            <p>&copy; {new Date().getFullYear()} All Rights Reserved. </p>
        </footer>
    )
}

export default Footer;
