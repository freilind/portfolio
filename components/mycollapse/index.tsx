import React, { FC } from "react";
import { Collapse, Grid } from "@nextui-org/react";
import { PortfolioList } from "../portfolio";
import { ExperienceList } from "../experience";
import Education from "../education/Education";
import { useIntl } from "react-intl";
import { CourseList } from "../courses";

const MyCollapse: FC = () => {
  const intl = useIntl();

  const title1 = intl.formatMessage({ id: "collapse.title.1" });
  const subtitle1 = intl.formatMessage({ id: "collapse.subtitle.1" });
  const title2 = intl.formatMessage({ id: "collapse.title.2" });
  const subtitle2 = intl.formatMessage({ id: "collapse.subtitle.2" });
  const title3 = intl.formatMessage({ id: "collapse.title.3" });
  const subtitle3 = intl.formatMessage({ id: "collapse.subtitle.3" });
  const title4 = intl.formatMessage({ id: "collapse.title.4" });
  const subtitle4 = intl.formatMessage({ id: "collapse.subtitle.4" });

  return (
    <Grid.Container gap={2}>
      <Grid>
        <Collapse.Group key={"collapse-group"}>
          <Collapse
            key={"collapse1"}
            title={title1}
            subtitle={subtitle1}
            aria-labelledby="collapse1"
          >
            <Grid.Container
              key={"collapse-container-1"}
              gap={4}
              justify="center"
            >
              <PortfolioList />
            </Grid.Container>
          </Collapse>
          <Collapse
            key={"collapse2"}
            title={title2}
            subtitle={subtitle2}
            aria-labelledby="collapse2"
          >
            <Grid.Container
              key={"collapse-container-2"}
              gap={2}
              justify="center"
            >
              <ExperienceList />
            </Grid.Container>
          </Collapse>
          <Collapse
            key={"collapse4"}
            title={title4}
            subtitle={subtitle4}
            aria-labelledby="collapse3"
          >
            <Grid.Container
              key={"collapse-container-4"}
              gap={4}
              justify="center"
            >
              <CourseList />
            </Grid.Container>
          </Collapse>
          <Grid
            xs={0}
            sm={12}
            css={{
              padding: "$0",
            }}
          >
            <Collapse
              key={"collapse3"}
              title={title3}
              subtitle={subtitle3}
              aria-labelledby="collapse4"
            >
              <Grid.Container
                key={"collapse-container-3"}
                gap={1}
                justify="center"
                xs={12}
              >
                <Education />
              </Grid.Container>
            </Collapse>
          </Grid>
        </Collapse.Group>
      </Grid>
    </Grid.Container>
  );
};

export default MyCollapse;
