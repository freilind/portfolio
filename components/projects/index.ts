import { IProject } from "../../interfaces";

export const projectsList: Array<IProject> = [
    {
        url: '/portfolio/github-user-search',
        title: 'Github user search',
        imgLight: '/images/github-search-light.png',
        imgDark: '/images/github-search-dark.png',
    },
    {
        url: '/portfolio/to-do',
        title: 'Todo list',
        imgLight: '/images/todo-light.png',
        imgDark: '/images/todo-dark.png'
    },
    {
        url: '/portfolio/markdown',
        title: 'Markdown',
        imgLight: '/images/markdown-preview-light.png',
        imgDark: '/images/markdown-preview-dark.png'
    },
    {
        url: '/portfolio/reduce-text',
        title: 'Reduce text',
        imgLight: '/images/reduce-light.png',
        imgDark: '/images/reduce-dark.png'
    },
    {
        url: '/portfolio/pokemon',
        title: 'Pokem√≥n API',
        imgLight: '/images/pokemon-light.png',
        imgDark: '/images/pokemon-dark.png'
    },
    {
        url: '/portfolio/fquiz',
        title: 'Fquiz App',
        imgLight: '/images/fquiz-light.png',
        imgDark: '/images/fquiz-dark.png'
    }
];
