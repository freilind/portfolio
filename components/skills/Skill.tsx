import React, { FC } from 'react';
import { ISkill } from '../../interfaces';

const Skill: FC<ISkill> = ({ tech }) => {
    return (
        <i className={`bx bx-lg bxl-${tech}`}></i>
    )
}

export default Skill;
