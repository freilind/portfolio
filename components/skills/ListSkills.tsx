import { FC } from "react";
import Skill from "./Skill";

const skills: Array<string> = [
  "java",
  "spring-boot",
  "javascript",
  "react",
  "tailwind-css",
  "aws",
  "mongodb",
];

const ListSkills: FC = () => {
  return (
    <>
      {skills.map((sk, i) => (
        <Skill tech={sk} key={i} />
      ))}
    </>
  );
};

export default ListSkills;
