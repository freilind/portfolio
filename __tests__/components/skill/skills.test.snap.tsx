import { render } from "@testing-library/react";
import { ListSkills } from "../../../components/skills";

describe("ListSkill snapshot", () => {
  it("renders skills unchanged", () => {
    const { container } = render(<ListSkills />);
    expect(container).toMatchSnapshot();
  });
});
