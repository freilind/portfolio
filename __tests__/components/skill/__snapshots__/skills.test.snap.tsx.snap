// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSkill snapshot renders skills unchanged 1`] = `
<div>
  <i
    class="bx bx-lg bxl-java"
  />
  <i
    class="bx bx-lg bxl-spring-boot"
  />
  <i
    class="bx bx-lg bxl-javascript"
  />
  <i
    class="bx bx-lg bxl-react"
  />
  <i
    class="bx bx-lg bxl-tailwind-css"
  />
  <i
    class="bx bx-lg bxl-aws"
  />
  <i
    class="bx bx-lg bxl-mongodb"
  />
</div>
`;
